include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
```

Comment:
This was one of my uncertainties, working on this — I don’t know what types are ever likely to pass through here. Should we be checking for larger numeric types in this function, or break out floats into a more specific `toString()` of its own?